importccl: allow COPY to import unordered or omitted columns

otan opened the pull request:
Refs: #52382

Release note (sql change): Previously, IMPORT PGDUMP using COPY required
all rows to be present in the same order as the table definition. This
restriction has been removed.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52488)
<!-- Reviewable:end -->

dt commented on pkg/ccl/importccl/read_import_pgdump.go:
```diff
@@ -680,17 +684,18 @@ func (m *pgDumpReader) readFile(
 				switch row := row.(type) {
 				case copyData:
-					if expected, got := len(conv.VisibleCols), len(row); expected != got {
+					if expected, got := len(i.Columns), len(row); expected != got {
```
should this be `len(conv.IsTargetCol)` that we compare to `len(row)`? or maybe `len(targetColMapInd)` ?

otan replied:
done

dt commented on pkg/ccl/importccl/read_import_pgdump.go:
```diff
@@ -648,14 +648,18 @@ func (m *pgDumpReader) readFile(
 			if importing && conv == nil {
 				return errors.Errorf("missing schema info for requested table %q", name)
 			}
+			var targetColMapInd []int
```
nit, here/below: don't we usually call this `idx` rather than `ind` ?

otan replied:
it was called `Ind` in the `INSERT INTO`, so I was form following. I can change both if you want.

otan replied:
done!

otan committed:
importccl: allow COPY to import unordered or omitted columns

Release note (sql change): Previously, IMPORT PGDUMP using COPY required
all rows to be present in the same order as the table definition. This
restriction has been removed.

otan commented:
tftr!

bors r=dt
